src/render/src/encoder.rs

```diff
@@ -135,13 +140,17 @@ impl<R: Resources, C: draw::CommandBuffer<R>> Encoder<R, C> {
     }
 
     /// Update a buffer with a single structure.
-    pub fn update_constant_buffer<T: Copy>(&mut self, buf: &handle::Buffer<R, T>, data: &T) {
+    pub fn update_constant_buffer<T: Copy>(&mut self, buf: &handle::Buffer<R, T>, data: &T) -> Result<(), UpdateError<usize>> {
```

Comment:
I think it's unnecessary to return `Result` here and propagate the error. In 99% of cases the user won't care. In that other 1% of cases, they may just use `update_buffer` instead, since this `update_constant_buffer` is just a helper wrapper (designed for convenience, after all).